### PR TITLE
bugfix for getPackagesByPackageName

### PR DESCRIPTION
**Pull request by tamarrow**
This fixes the failing listVersions test in `PackageDescribeSpec`

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala [hunk 1]
```diff
@@ -30,7 +30,9 @@ final class MultiRepository(
         }
       } map (_.flatten)
     } map { packages =>
-      packages.headOption.getOrElse(throw PackageNotFound(packageName))
+      packages
+        .filterNot(_.isEmpty)
+        .headOption.getOrElse(throw PackageNotFound(packageName))
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: How about `.find(!_.isEmpty).getOrElse(...)`?

Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: done

**Comment by jsancio** [comment]
LGTM! 🚢 it.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala [hunk 2]
```diff
@@ -30,7 +30,9 @@ final class MultiRepository(
         }
       } map (_.flatten)
     } map { packages =>
-      packages.headOption.getOrElse(throw PackageNotFound(packageName))
+      packages
+        .find(!_.isEmpty)
+        .getOrElse(throw PackageNotFound(packageName))
     }
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: I'm wondering if we should have this code instead?

```scala
repositories().flatMap { repositories =>
  Future.collect {
    // ...
  } map (_.flatten.flatten.toList)
}
```

Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: I verified locally that this change makes `PackageDescribeSpec` pass.

Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: Well, this is not how the old Cosmos works. I am okay changing it but let's not do it in this PR and let's talk about the consequences before doing it.

Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/MultiRepository.scala]: Merging into @cruhland's branch as is, and we can always change it later.

**Pull request merged**